src/main/webapp/profLoader.js

```diff
@@ -0,0 +1,128 @@
+/* using IIFE https://flaviocopes.com/javascript-iife/ */
+(() => {
+  google.charts.load("current", { packages: ["corechart"] });
+  google.charts.load("current", { packages: ["bar"] });
+
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const profKey = urlParams.get("prof-key");
+  const profName = urlParams.get("prof-name");
+
+  function populateData() {
+    fetch(`/prof-data?prof-key=${profKey}`)
+      .then((response) => response.json())
+      .then((profInfo) => {
+        document.getElementById("prof-name").innerHTML = profName;
+        google.charts.setOnLoadCallback(() => {
+          makeDiffComparisonChart(profInfo);
+          makePerceptionComparisonChart(profInfo);
+        });
+      });
+  }
+
+  document.addEventListener("DOMContentLoaded", function () {
+    populateData();
+  });
+
+  function makeDiffComparisonChart(profData) {
+    const average = (list) =>
+      list.reduce((prev, curr) => prev + curr, 0) / list.length;
+
+    const courseList = [];
+    const difficultyAvgList = [];
+
+    for (let dataHolder of profData) {
+      courseList.push(dataHolder.term);
+      difficultyAvgList.push(
+        average(dummyGradeData.concat(dataHolder.difficultyList)) + 3
+      );
+    }
+
+    const difficultyData = google.visualization.arrayToDataTable([
+      [" "].concat(courseList),
+      [" "].concat(difficultyAvgList),
+    ]);
+
+    const options = {
+      colors: ["#81b8ec", "#f1a79d", "#f1d19d"],
+      title: "Average Course Difficulty Comparison",
+      height: 450,
+      bars: "horizontal",
+      bar: { groupWidth: "30%" },
+      hAxis: { title: "Avg Term Perception" },
+    };
+
+    const chart = new google.charts.Bar(
+      document.getElementById("difficulty-comp")
+    );
+    chart.draw(difficultyData, google.charts.Bar.convertOptions(options));
+  }
+
+  function makePerceptionComparisonChart(profData) {
+    const average = (list) =>
+      list.reduce((prev, curr) => prev + curr, 0) / list.length;
+
+    const courseList = [];
+    const perceptionAvgList = [];
+    const commentList = [];
+
+    for (let dataHolder of profData) {
+      courseList.push(dataHolder.term);
+      perceptionAvgList.push(
+        average(dummyGradeData.concat(dataHolder.perceptionList)) + 7
```

Comment:
Same question as above, why 7 here? Comment would help

Reply:
will add, its just to make the dummy data look different from the other data